### PR TITLE
doc: board_porting: Provide guideline on MPU activation

### PR DESCRIPTION
**Pull request by erwango**
Provide a rule to settle on MPU activation by default.
Request is to activate it by default if board resources allow.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

### CODE REVIEW

File: doc/guides/porting/board_porting.rst [hunk 1]
```diff
@@ -372,6 +372,9 @@ while porting.
 
 - If available, enable pinmux and interrupt controller drivers.
 
+- When possible enable MPU. For boards with limited resources, it is acceptable
+  to let it disabled.
```
Comment (mbolivar) [inline doc/guides/porting/board_porting.rst]:
```suggestion
  to disable it.
```

**Review by mbolivar — APPROVED** [review]

File: doc/guides/porting/board_porting.rst (hunk 1 above)
Comment (erwango) [reply doc/guides/porting/board_porting.rst]: Updated, txs!

**Review by mbolivar — APPROVED** [review]
Seems reasonable as long as this is not eventually going to be a recommendation to enable userspace by default.

**Review by ioannisg — COMMENTED** [review]
fine with the message it conveys, I would reword this as:

> For ARM Cortex-M boards, it is recommended to enable the MPU by default, if there is support for it in hardware. For boards with limited memory resources it is acceptable to not enabled it.

**Comment by ioannisg** [comment]
> Seems reasonable as long as this is not eventually going to be a recommendation to enable userspace by default.

User space is a different story. With basic MPU we are able to catch lots or errors related to accessing restricted memory areas, executing from RAM on XIP, stack overflows, etc., so it is very useful, and the footprint is limited.

**Comment by mbolivar** [comment]
> to not enabled it.

"to disable it" if you are going this route

**Comment by ioannisg** [comment]
> > to not enabled it.
> 
> "to disable it" if you are going this route

Agreed

**Comment by erwango** [comment]
@ioannisg 

> "For ARM Cortex-M boards" 

Is that really restricted to this specific arch ? Don't other archs have similar features?

**Comment by ioannisg** [comment]
> @ioannisg
> 
> > "For ARM Cortex-M boards"
> 
> Is that really restricted to this specific arch ? Don't other archs have similar features?

I believe so, yes; x86 and high-end Cortex-A,R have MMU, which is different from MPU.
ARC does have MPU but such boards should not be accounted as low-end devices, so the comment should not be applicable, IMHO.

**Comment by erwango** [comment]
@ioannisg 

> ARC does have MPU but such boards should not be accounted as low-end devices

So, MPU enabling recommendation is not limited to Cortex-M. Only the on resources basis disabling applies to Cortex-M. Which is true today, but some time later a new type of arch can show up with similar issues, isn't it ? I don't see the gain in not making this more generic.

**Comment by ioannisg** [comment]
>  Which is true today, but some time later a new type of arch can show up with similar issues, isn't it ? 

Well, I would only think of low-end RISC-V platforms, adding MPU support in the future; then the comment could apply for such platforms. It's fine with me if you want to make the sentence more generic (but it definitely applies only to cortex-m, atm).

**Comment by erwango** [comment]
> Well, I would only think of low-end RISC-V platforms, adding MPU support in the future; then the comment could apply for such platforms. It's fine with me if you want to make the sentence more generic (but it definitely applies only to cortex-m, atm).

Thanks. Would you mind approving then ?

**Comment by ioannisg** [comment]
Sure, I'd still want to hear opinions from other ARM board maintainers, though. FYI @MaureenHelm @galak :)

**Review by ioannisg — APPROVED** [review]

**Comment by ioannisg** [comment]
@erwango perhaps you can still update based on my recommendation above (excluding the part about cortex-m). Makes it more clear about what sort of limitations we consider :)

**Comment by erwango** [comment]
> @erwango perhaps you can still update based on my recommendation above (excluding the part about cortex-m). Makes it more clear about what sort of limitations we consider :)

Sure, Done!
